db/functions/function.go

```diff
@@ -235,7 +235,7 @@ func (fn *functionImpl) authorize(user auth.User, args map[string]any) error {
 				return nil
 			} else if channel, err := allow.expandPattern(channelPattern, args, user); err != nil {
 				return err
-			} else if user.CanSeeChannel(channel) {
+			} else if user.CanSeeCollectionChannel(base.DefaultScope, base.DefaultCollection, channel) {
```

Comment:
You should file a ticket for jens to fix this, graphql has some collections awareness now.

Reply:
Filed https://issues.couchbase.com/browse/CBG-2640